.github/workflows/golang.yaml

```diff
@@ -0,0 +1,24 @@
+name: GoLang CI
+on:
+  push:
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Dependencies
+        run: chmod 777 -R "$(pwd)" && make dep
+      - name: Lint
+        run: make lint
+      - name: Unit Tests
+        run: make test
+      - name: Integration Tests
+        run: make integration
+      - name: Gather Coverage
+        run: make coverage
+      - name: Upload Coverage
+        continue-on-error: true
+        run: bash <(curl -s https://codecov.io/bash) -f .coverage/combined.cover.out
```

Comment:
Question: Can you refresh me on where we stand with codecov? I know we eradicated it from our private repos, is it something we're fine with for public ones since there's no secrets we're worried about?

Happy with this move from Travis either way though!

Reply:
I am still thinking about what to do with codecov in github. Technically there is no threat as we do not have *any* secrets in CI and should not publish any artifacts  (we do not use codecov where we do any of that). Sonar has [free Cloud product for OSS repos](https://www.sonarsource.com/open-source-editions/#:~:text=SonarCloud%20is%20entirely%20free%20for,want%20to%20analyze%20private%20repositories.) , which is probably the way to go from codecov.